### DISCUSSION TITLE
Empty promise returned useQuery refetch?

### DISCUSSION THREAD

**Discussion by hainessss** [TanStack/query#1718] [Q&A]
Hey all, 

Looking for guidance. I am confused as to why the `refetch` function from useQuery returns an empty promise. I would expect that data from the query would be returned here.

```
const data = await refetch();

// data is undefined
```

My use case is to reset a form's state on submit. Something like this.

```
 await mutate(formState)

const newData = await refetch()

setForm({ ...newData })
```

How are you all doing this? I am I thinking about this wrong?

I can't think of reason to not return the data from refetch?

Help is appreciated, thanks.

**Comment by TkDodo** [comment]
according to the docs, it should return a `Promise<UseQueryResult>`, so inside the returned result, there should be `.data`. It's basically the same structure that `useQuery` returns.

**Reply by hainessss** [reply]
hmm, I am getting `undefined`.

**Reply by TkDodo** [reply]
Can you create a codesandbox example that shows your situation?

**Comment by marinvirdol** [comment]
I've just tried out `refetch` function from version `3.6.0` and for me it returned the data.

I think you could achieve the same result by invalidating the query `onSuccess` and then having a `useEffect` to set the form whenever query data changes.
